File: RodRo-Server/src/main/java/cz/rodro/constant/ConfessionType.java
package cz.rodro.constant;

import lombok.Getter;

@Getter
public enum ConfessionType {

    CATHOLIC_LATIN("Roman Catholic (Latin)"),
    CATHOLIC_UNIATE("Greek Catholic / Uniate"),
    ORTHODOX("Eastern Orthodox"),
    PROTESTANT_LUTHERAN("Lutheran / Evangelical"),
    PROTESTANT_REFORMED("Calvinist / Reformed"),
    ANGLICAN("Anglican"),
    JEWISH("Jewish"),
    ISLAMIC("Islamic (Muslim)"),
    OTHER("Other / Unknown");

    // Field to hold the human-readable name
    private final String displayName;

    /**
     * Constructor for the enum, setting the display name.
     * @param displayName The user-friendly name of the confession type.
     */
    ConfessionType(String displayName) {
        this.displayName = displayName;
    }

    /**
     * Getter method required by the MapStruct mapper.
     * @return The user-friendly display name.
     */
    public String getDisplayName() {
        return displayName;
    }
}

File: RodRo-Client/src/constants/confessionType.js
/**
 * CONFESSION_TYPE_MAP
 * Maps backend ConfessionType enum keys (uppercase_with_underscores) to 
 * user-friendly display names for the frontend.
 */
export const CONFESSION_TYPE_MAP = { 
    // Key (from API) : Value (for display)
    CATHOLIC_LATIN: 'Roman Catholic (Latin)',       
    CATHOLIC_UNIATE: 'Greek Catholic / Uniate',     
    ORTHODOX: 'Eastern Orthodox', // <--- This key is required to match the API's enum name
    PROTESTANT_LUTHERAN: 'Lutheran / Evangelical',  
    PROTESTANT_REFORMED: 'Calvinist / Reformed',    
    ANGLICAN: 'Anglican',                           
    JEWISH: 'Jewish',                               
    ISLAMIC: 'Islamic (Muslim)',                    
    OTHER: 'Other / Unknown'                        
};